metaflow/runtime.py

```diff
@@ -606,7 +650,13 @@ def _launch_worker(self, task):
         for fd in worker.fds():
             self._workers[fd] = worker
             self._poll.add(fd)
-        self._num_active_workers += 1
+        active_step_counts = self._active_tasks.setdefault(task.step, [0, 0])
+
+        # We have an additional task for this step running
+        active_step_counts[0] += 1
```

Comment:
Where is this used? We can set the defaults as `[1,0]` directly?

Reply:
We could initialize to [1, 0] but then would have to be a different path for doesn't exist and exists. Here, the active_step_counts always points to what needs to be updated. It's used to print the number of tasks running/done.

Reply:
How is replacing with `self._active_tasks.setdefault(task.step, [1, 0])` necessitating a different path?

Reply:
self._active_tasks[task.step][0] by adding one whether or not task.step is already in self._active_tasks. The above code is equivalent to:
```
if task.step in self._active_tasks:
  self._active_tasks[task.step][0] += 1
else:
  self._active_tasks[task.step] = [1, 0]
```
with the two paths. The code simplifies that a bit IMHO.